web/scripts/users/profileComponent.js

```diff
@@ -0,0 +1,102 @@
+/* global alert */
+module.exports = {
+  scope: 'heinz',
+  name: 'profileComponent',
+  dependencies: ['router', 'Vue', 'usersRepo', 'storage'],
+  factory: (router, Vue, usersRepo, storage) => {
+    'use strict'
+
+    var state = { storage, newcat: null }
+    const component = Vue.component('profile', {
+      template: `
+      <div id="user-profile">
+        <div class="form-group">
+          <label>Full Name</label></br>
+          {{storage.get('user').name}}
+        </div>
+        <div class="form-group">
+          <label>Email</label></br>
+          {{storage.get('user').email}}
+        </div>
+        </br>
+        <div class="form-group">
+          <label>Categories</label></br>
+          <table class="category-table">
+            <tr v-for="cat in storage.get('user').categories">
+              <td class="td">
+                {{cat}}
+              </td>
+              <td class = "td">
+              <button class="btn btn btn-danger btn-sm" v-on:click="removeCat(cat)">Remove</button>
+              </td>
+            </tr>
+            <tr>
+              <td class = "td"><input v-model="newcat" name="newcat" class="form-control form-control-sm" type="text" placeholder="New Category"></td>
+              <td class = "td">
+              <button class="btn btn btn-success btn-sm" v-on:click="addCat">Add</button>
+              </td>
+            </tr>
+          </table>
+        </div>
+        <br/>
+        <br/>
+        <button class="btn btn-primary" v-on:click="logout">Logout</button>
+        <br />
+        <br />
+      </div>
+    `,
+      data: function () {
+        return state
+      },
+      methods: {
+        logout: function (event) {
+          storage.clear()
+          document.getElementById("username-view").innerHTML = ''
+
+          return router.navigate('/')
+        },
+
+        removeCat: function (cat) {
+          let user = storage.get('user')
+
+          usersRepo.remCat(user.email, cat, (err, res) => {
+            if (err) {
+              alert('Unable to remove category: check connection')
+              return
```

Comment:
There are eslint errors in this file as well.

Reply:
:+1